Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift

```diff
@@ -8,134 +8,338 @@
-			
-			return true
 		case .failure(let error):
-			Debug.DLog("peerGetClientKey exception: \(error)")
+			Debug.DLog("initSessionUserPeer GetClientKey fail with exception: \(error)")
+			return false
+		}
+	}
+	
+	func registerSenderKeyToGroup(groupID: Int64, clientId: String, domain: String) async -> Bool {
+		do {
+			let identityKey = try signalStore.identityStore.identityKeyPair(context: NullContext())
+			let privateKey = identityKey.privateKey
+			
+			let senderAddress = try ProtocolAddress(name: "\(domain)_\(clientId)", deviceId: UInt32(Constants.senderDeviceId))
+			let distributionId = senderStore.getSenderDistributionID(from: senderAddress, groupId: groupID)
+			let sentAliceDistributionMessage = try SenderKeyDistributionMessage(from: senderAddress, distributionId: distributionId, store: senderStore, context: NullContext())
+			
+			let pbkdf2 = PBKDF2(passPharse: bytesConvertToHexString(bytes: privateKey.serialize()))
+			
+			let key = KeyHelper.generateIdentityKeyPair()
+			let encryptedGroupPrivateKey = pbkdf2.encrypt(data: key.privateKey.serialize(), saltHex: "") // retrieve saltHex from database
+			
+			let senderKey = try senderStore.loadSenderKey(from: senderAddress, distributionId: distributionId, context: NullContext())
+			
+			var request = Signal_GroupRegisterClientKeyRequest()
+			request.privateKey = bytesConvertToHexString(bytes: encryptedGroupPrivateKey ?? [])
+			request.groupID = groupID
+			request.deviceID = Int32(Constants.senderDeviceId)
+			request.publicKey = Data(key.publicKey.serialize())
+			request.clientKeyDistribution = Data(sentAliceDistributionMessage.serialize())
+			request.senderKey = Data(senderKey?.serialize() ?? [])
+			request.senderKeyID = Int64(distributionId.uuidString) ?? 0
+			
+			guard let server = channelStorage.realmManager.getServer(by: domain) else { return false }
+			let response = await channelStorage.getChannel(domain: server.serverDomain, accessToken: server.accessKey, hashKey: server.hashKey).groupRegisterClientKey(request)
+			switch response {
+			case .success(let result):
+				return true
+			case .failure(let error):
+				return false
+			}
+		} catch {
 			return false
 		}
 	}
 	
-	func processKeyStoreHasPrivateKey(recipientResponse: Signal_PeerGetClientKeyResponse) {
-		if let ourEncryptionMng = CKSignalCoordinate.shared.ourEncryptionManager {
+	func decryptPeerMessage(senderName: String, message: Data) -> String? {
+		do {
+			let signalProtocolAddress = try ProtocolAddress(name: senderName, deviceId: UInt32(Constants.receiverDeviceId))
+			let preKeyMessage = try PreKeySignalMessage(bytes: message)
+			let decryptMessage = try signalDecryptPreKey(
+				message: preKeyMessage, from: signalProtocolAddress, sessionStore:
+				signalStore.sessionStore, identityStore: signalStore.identityStore,
+				preKeyStore: signalStore.preKeyStore,
+				signedPreKeyStore: signalStore.signedPreKeyStore, context: NullContext())
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/88c1f4f9b0f08387d7d5bf2f7e72ae38123bc3b5/Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift#L269"></span>Arguments should be either on the same line, or one per line.<br />
<code>multiline_arguments</code> <code>MessageService.swift:269</code></td>
    </tr>
  </tbody>
</table>